process/smartContract/process.go

```diff
@@ -942,23 +992,35 @@ func fillWithESDTValue(fullVMInput *vmcommon.ContractCallInput, newVMInput *vmco
 	newVMInput.ESDTValue = big.NewInt(0).SetBytes(fullVMInput.Arguments[1])
 }
 
-func (sc *scProcessor) isCrossShardESDTTransfer(tx data.TransactionHandler) bool {
+func (sc *scProcessor) isCrossShardESDTTransfer(tx data.TransactionHandler) (string, bool) {
 	sndShardID := sc.shardCoordinator.ComputeId(tx.GetSndAddr())
 	if sndShardID == sc.shardCoordinator.SelfId() {
-		return false
+		return "", false
 	}
 
 	dstShardID := sc.shardCoordinator.ComputeId(tx.GetRcvAddr())
 	if dstShardID == sndShardID {
-		return false
+		return "", false
 	}
 
-	function, _, err := sc.argsParser.ParseCallData(string(tx.GetData()))
+	function, args, err := sc.argsParser.ParseCallData(string(tx.GetData()))
 	if err != nil {
-		return false
+		return "", false
 	}
 
-	return function == core.BuiltInFunctionESDTTransfer
+	if function != core.BuiltInFunctionESDTTransfer {
+		return "", false
+	}
+	if len(args) < 2 {
```

Comment:
Is this a case of backwards compatibility?

Reply:
yes it is. this code was not enabled yet.